jax/_src/lax/control_flow.py

```diff
@@ -315,6 +316,9 @@ def _create_jaxpr(init_val):
     new_init_val, = tree_unflatten(in_tree, new_init_vals)
     init_vals, init_avals, body_jaxpr, in_tree, *rest = _create_jaxpr(new_init_val)
   cond_jaxpr, cond_consts, body_consts, body_tree = rest
+  joined_effects = core.join_effects([body_jaxpr.effects, cond_jaxpr.effects])
+  body_jaxpr = body_jaxpr.replace(jaxpr=body_jaxpr.jaxpr.replace(effects=joined_effects))
+  cond_jaxpr = cond_jaxpr.replace(jaxpr=cond_jaxpr.jaxpr.replace(effects=joined_effects))
```

Comment:
Also mentioned below: let's just raise a NotImplementedError if any control flow primitives get jaxprs with effects in them (in the traceables, maybe also in the typecheck rules). Getting these right will take some care so we should probably add support for control flow primitives one-by-one in follow-up PRs.

To put a fine point on it: I think the only changes this PR should make to control_flow.py should be to raise NotImplementedErrors (if this PR makes any changes at all).

Reply:
Done.